### PR TITLE
[ML] Add SonarQube Scanning for Pull Requests

### PR DESCRIPTION
**Pull request by valeriy42**
This PR integrates SonarQube scanning for Pull Requests, enabling automatic code quality checks on every new or updated PR. SonarQube will analyze changes for code smells, bugs, and security vulnerabilities, providing feedback directly in the Pull Request. This helps maintain code quality standards and identify issues early.

Some details:
- SonarQube scans all new PRs.
- The changes are compared to the baseline of the `main` branch
- Comments with detailed feedback are added directly to the PR.
- At the moment, PRs don't have to meet code quality standards to pass (soft fail).
- Analysis runs within the existing CI/CD pipeline.
- There is no code coverage information.
- Unless the quality rules are changed, most files will be cached so that the analysis overhead is minimal (<3min)

### CODE REVIEW

File: .buildkite/pipelines/sonarqube.yml.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - github_commit_status:
+          context: "Generate compile_commands.json"
+  - label: "Static code analysis with SonarQube :sonarqube:"
+    key: "sonarqube"
+    depends_on: "export_compile_commands"
+    env:
+      VAULT_SONAR_TOKEN_PATH: "secret/ci/elastic-ml-cpp/sonar-analyze-token"
+    command: 
+      - "buildkite-agent artifact download cmake-build-docker/compile_commands.json ."
+      - ".buildkite/scripts/steps/sonar-qube.sh"
```
Comment (edsavage) [inline .buildkite/pipelines/sonarqube.yml.sh:32]: I _think_ this script has been missed from the PR?

Comment (valeriy42) [reply .buildkite/pipelines/sonarqube.yml.sh]: Indeed. It should be "/scan-source-code.sh" which is available in the sonar docker image.

**Comment by valeriy42** [comment]
@edsavage I have reconfigured the scanner to run on PR, I think having a report on every change will be most beneficial. Once the quality profile is stabilized, we can make it gating. 

It would be great if you could take another look.

**Review by edsavage — APPROVED** [review]
LGTM

Thanks for doing this Valeriy!

**Pull request merged**